iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/helplets.inl

```diff
@@ -148,6 +143,9 @@ inline bool doesEndWithPathSeparator(const string<StringCapacity>& name) noexcep
 
     for (uint64_t i = 0; i < iox::platform::IOX_NUMBER_OF_PATH_SEPARATORS; ++i)
     {
+        // in platform the path separator and number of path separators are set and tested
+        // so that such an access is safe
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
         if (lastCharacter == iox::platform::IOX_PATH_SEPARATORS[i])
         {
             return true;
```

Comment:
This could also be a range based for loop like below. This makes the linter suppression obsolete.